tgui/packages/tgui/interfaces/Oven.tsx

```diff
@@ -0,0 +1,218 @@
+                      onClick={() => act('eject', { ejected_item: index + 1 })}
+                      ml={3}
+                      style={{ transform: 'translate(0, 4px)' }}
+                    >
+                      <Icon name="eject" />
+                    </Button>
+                  </Stack.Item>
+                </Stack>
+              ))}
+            </Stack>
+          ) : (
+            <Stack minHeight="110px" maxHeight="110px"><Stack.Item>(Empty)</Stack.Item></Stack>
+          )}
+        </Section>
+        {output_icon && (
+          <Section grow>
+            <Stack align>
+              <Stack.Item grow>
+                <Section title="Potential Recipe">
+                  <Stack vertical>
+                    <Stack.Item>
+                    {recipe_icons.map((item, index) => (
+                      <Stack.Item key={index} style={{ borderBottom: '0.5px #555 solid', display: "flex" }}>
+                        <img
+                          height={26}
+                          width={26}
+                          src={`data:image/png;base64,${item}`}
+                          style={{ transform: 'translate(0, -4px)' }}
+                        />
+                        {recipe_names[index]}
+                      </Stack.Item>
+                    ))}
+                    </Stack.Item>
+                    <Stack.Item>
+                      <LabeledList>
+                        <LabeledList.Item label="Cook time">
+                          {cook_time}
+                        </LabeledList.Item>
+                      </LabeledList>
+                    </Stack.Item>
+                  </Stack>
+                </Section>
+              </Stack.Item>
+              <Stack.Item><Divider vertical /></Stack.Item>
+              <Stack.Item grow>
+                <Section title="Result">
+                  <Stack vertical align="center" textAlign="center">
+                    <Stack.Item>
+                      <img
+                        height={48}
+                        width={48}
+                        src={`data:image/png;base64,${output_icon}`}
+                        style={{ transform: 'translate(0, -4px)' }}
+                      />
+                    </Stack.Item>
+                    <Stack.Item>
+                     {output_name}
+                    </Stack.Item>
+                  </Stack>
+                </Section>
+              </Stack.Item>
+            </Stack>
+          </Section>
+        )}
+        {!!cooking && (
+          <Modal fontSize={2} textAlign="center">
+            <Section>Cooking! Please wait...</Section>
+          </Modal>
+        )}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const OvenDialbuttons = (props) => {
+  const { act } = useBackend();
+  const { min, max, time } = props;
+  const nodes: JSX.Element[] = [];
+  for (let i = min; i <= max; i++) {
+    const node = (
+      <Button key={i}
+        color={time === i ? 'default' : 'grey'}
```

Comment:
Grey means disabled, buttons should use the default blue for clickable and green for selected.